pkg/scheduler/worker/workerpool_test.go

```diff
@@ -17,16 +20,25 @@ import (
 )
 
 type testInvoker struct {
-	params []*invoker.Params
+	params     []*invoker.Params
+	reconRepo  reconciliation.Repository
+	mux        sync.Mutex
+	errChannel chan error
 }
 
 func (i *testInvoker) Invoke(_ context.Context, params *invoker.Params) error {
+	if err := i.reconRepo.UpdateOperationState(params.SchedulingID, params.CorrelationID, model.OperationStateInProgress, ""); err != nil {
+		i.errChannel <- errors.New("Update failed")
+		return err
+	}
+	i.mux.Lock()
 	i.params = append(i.params, params)
+	i.mux.Unlock()
 	return nil
 }
 
 func TestWorkerPool(t *testing.T) {
-	test.IntegrationTest(t) //required because a valid Kubeconfig is required to create test cluster entry
+	//test.IntegrationTest(t) //required because a valid Kubeconfig is required to create test cluster entry
```

Comment:
As far as I understood, this function is a helper to control if needed to run it as an integration test. Although the way how we handle the distinguish between unit and integration test can be challenged, to remove it is not a good idea at the moment.

Reply:
Was removed by fault, will add it again